validator-api/src/coconut/client.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright 2022 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::coconut::error::{CoconutError, Result};
+use crate::config::DEFAULT_LOCAL_VALIDATOR;
+
+use validator_client::nymd::{tx::Hash, NymdClient, QueryNymdClient, TxResponse};
+
+use async_trait::async_trait;
+
+#[async_trait]
+pub trait Client {
+    async fn get_tx(&self, tx_hash: &str) -> Result<TxResponse>;
+}
+
+pub struct QueryClient {
```

Comment:
The idea is that I wrap the client in this so I can implement the trait specific to what I'm using it for. And the trait is useful so that I can create a mock client in unit tests and not depend on the nymd chain running. I really think we should expand this approach to the entire nymd client and enable more stuff to be tested in the codebase, but maybe we can have a talk about this with the others.